### PR TITLE
היי לוצקי, מימשתי lexical density, תמזג אצלך

### PR DESCRIPTION
**Pull request by gal-star**
היי לוצקי, מימשתי lexical density, תמזג אצלך

### CODE REVIEW

**Comment by lutzky** [comment]
Please implement the actual hypotheses in separate modules. See for example [`translationese/contractions.py`](https://github.com/lutzky/translationese/blob/master/translationese/contractions.py).

**Comment by lutzky** [comment]
Oh, you're right. My mistake.
Can you separate it into its own module and merge my own changes so this merges cleanly?

Implementing as a separate module allows usage of the `analyze.py` utility which generates the ARFF files.

**Comment by gal-star** [comment]
What do u mean "seperate modules" - you mean different files?

**Comment by lutzky** [comment]
Yes, again - see the example I linked.

**Comment by lutzky** [comment]
You can see that here the Travis build still fails... please merge my change to `.travis.yml`.

**Comment by lutzky** [comment]
This looks better now, but why did you return `mean_word_length` and `mean_sentence_length` to `translationese/__init__.py`? Those were moved to external modules on purpose.

File: translationese/lexical_density.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+def quantify(analysis):
+    is_not_verb = lambda letter: letter !='V'
+    is_not_noun = lambda letter: letter !='N'
+    is_not_adjective = lambda letter: letter !='J'
+    is_not_adverb = lambda letter: letter !='R'
+        
+    is_lexical_density = lambda l: is_not_verb(l) and is_not_noun(l) and \
+            is_not_adjective(l) and is_not_adverb(l)
+    
+
```
Comment (lutzky) [inline translationese/lexical_density.py:34]: Extra space

File: translationese/lexical_density.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    is_not_noun = lambda letter: letter !='N'
+    is_not_adjective = lambda letter: letter !='J'
+    is_not_adverb = lambda letter: letter !='R'
+        
+    is_lexical_density = lambda l: is_not_verb(l) and is_not_noun(l) and \
+            is_not_adjective(l) and is_not_adverb(l)
+    
+    
+    return float(len([x for (x,y) in analysis.pos_tags() if is_lexical_density(y[0])])) / len(analysis.tokens())
+
```
Comment (lutzky) [inline translationese/lexical_density.py]: Extra space

File: translationese/lexical_density.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    is_not_verb = lambda letter: letter !='V'
+    is_not_noun = lambda letter: letter !='N'
+    is_not_adjective = lambda letter: letter !='J'
+    is_not_adverb = lambda letter: letter !='R'
+        
+    is_lexical_density = lambda l: is_not_verb(l) and is_not_noun(l) and \
+            is_not_adjective(l) and is_not_adverb(l)
+    
+    
+    return float(len([x for (x,y) in analysis.pos_tags() if is_lexical_density(y[0])])) / len(analysis.tokens())
```
Comment (lutzky) [inline translationese/lexical_density.py]: Overly complex statement, needs split. (Generate a list, then get its average)

File: tests/test_lexical_density.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+class LexcialDensityTest(unittest.TestCase):
+    def test_simple(self):
+        a1 = translationese.Analysis("""Hello hello world world""")
+        a2 = translationese.Analysis("""I'm certain he said I am later than I am, 
+                                     let's go""")
+        a3 = translationese.Analysis("""we will go tomorrow to see a movie""")
+        a4 = translationese.Analysis("""What about punctuation?""")
+        
+        self.assertEqual(0.0, lexical_density.quantify(a1))
```
Comment (lutzky) [inline tests/test_lexical_density.py]: `[module].quantify` must return a `dict: attribute -> float`, or else it won't work with `analyze.py`.

**Pull request merged**